### ISSUE TITLE
better experiments

### ISSUE THREAD

**Issue by yuchenhou** [yuchenhou/elephant#47]
- more exploration in parameter space
- average runs to reduce the effect of random units initialization

**Comment by yuchenhou** [comment]
![error_vs_num_epochs_forum](https://user-images.githubusercontent.com/8304443/30313887-c54ec098-9753-11e7-80ce-35bc94479bc6.png)
![error_vs_num_epochs_congress](https://user-images.githubusercontent.com/8304443/30313888-c54f213c-9753-11e7-9ea9-99d003496ac4.png)
![error_vs_num_epochs_collaboration](https://user-images.githubusercontent.com/8304443/30313889-c55017b8-9753-11e7-8446-ae64d92817ca.png)
![error_vs_num_epochs_airport](https://user-images.githubusercontent.com/8304443/30313891-c6231fb4-9753-11e7-8f45-bd4fd25413c3.png)

**Comment by lbholder** [comment]
Strange behavior on the Forum dataset. Any thoughts as to why?

**Comment by yuchenhou** [comment]
Look for correlations between optimum parameters and graph properties.

**Comment by yuchenhou** [comment]
![error_vs_dimension_forum](https://user-images.githubusercontent.com/8304443/30572249-529509ba-9ca1-11e7-95d1-4c595b5573e9.png)
![error_vs_dimension_congress](https://user-images.githubusercontent.com/8304443/30572251-5296cc5a-9ca1-11e7-8764-b62161cf0d64.png)
![error_vs_dimension_collaboration](https://user-images.githubusercontent.com/8304443/30572250-5296833a-9ca1-11e7-9491-09743dd7a713.png)
![error_vs_dimension_airport](https://user-images.githubusercontent.com/8304443/30572248-52932dde-9ca1-11e7-9a32-3a5f13843a65.png)
![average_degree](https://user-images.githubusercontent.com/8304443/30572258-5d00a7e2-9ca1-11e7-83e2-a85195ffc688.png)
![number_of_edges](https://user-images.githubusercontent.com/8304443/30572259-5d01aa3e-9ca1-11e7-91c0-d619f19778a6.png)
![number_of_nodes](https://user-images.githubusercontent.com/8304443/30572257-5d0040fe-9ca1-11e7-804e-0a3cff4be9f3.png)

**Comment by lbholder** [comment]
What is the x-axis on the error plots? "dimension"?

**Comment by yuchenhou** [comment]
It's the number of units in a layer, the layer size, the network width. Which of these do you prefer?

**Comment by lbholder** [comment]
How about "Units per Layer"? If there are multiple layers, do they all have the same number of units?

In terms of effect on error, looks like higher numbers are better, though not a big difference in most datasets.

**Comment by yuchenhou** [comment]
> If there are multiple layers, do they all have the same number of units?

Yeah

> How about "Units per Layer"?

Yeah I changed to this new name.

**Comment by yuchenhou** [comment]
![num_hidden_layers_facebook](https://user-images.githubusercontent.com/8304443/30779258-b46cc58a-a0a0-11e7-9d4e-ff11a4e80c5f.png)
![num_hidden_layers_authors](https://user-images.githubusercontent.com/8304443/30779256-b465a39a-a0a0-11e7-82bb-fbf3caca92ed.png)
![num_hidden_layers_forum](https://user-images.githubusercontent.com/8304443/30779257-b4681a8a-a0a0-11e7-9437-c56215e7fba6.png)
![num_hidden_layers_congress](https://user-images.githubusercontent.com/8304443/30779253-b4648eec-a0a0-11e7-8344-2c1666f2a664.png)
![num_hidden_layers_collaboration](https://user-images.githubusercontent.com/8304443/30779254-b46557d2-a0a0-11e7-81a2-072d51fe9f4c.png)
![num_hidden_layers_airport](https://user-images.githubusercontent.com/8304443/30779255-b4656e02-a0a0-11e7-828d-690253fc0bed.png)

**Comment by yuchenhou** [comment]
Add training time vs data set size measure

**Comment by yuchenhou** [comment]
![running_time](https://user-images.githubusercontent.com/8304443/31113552-b3b05a28-a7ce-11e7-9e19-728a7571ac69.png)

**Comment by yuchenhou** [comment]
I added improved experiments into the paper.
